tools/pythonpkg/src/vector_conversion.cpp

```diff
@@ -410,6 +410,57 @@ void VectorConversion::NumpyToDuckDB(PandasColumnBindData &bind_data, py::array
 	}
 }
 
+void VectorConversion::BindNumpy(const DBConfig &config, py::handle df, vector<PandasColumnBindData> &bind_columns,
+                                 vector<LogicalType> &return_types, vector<string> &names) {
+	auto df_columns = py::list(df.attr("keys")());
+	auto df_types = py::list();
+	for (auto item : py::cast<py::dict>(df)) {
+		if (string(py::str(item.second.attr("dtype").attr("char"))) == "U") {
+			df_types.attr("append")(py::str("string"));
+			continue;
+		}
+		df_types.attr("append")(item.second.attr("dtype"));
+	}
+	auto get_fun = df.attr("__getitem__");
+	if (py::len(df_columns) == 0 || py::len(df_types) == 0 || py::len(df_columns) != py::len(df_types)) {
+		throw InvalidInputException("Need a DataFrame with at least one column");
+	}
+	for (idx_t col_idx = 0; col_idx < py::len(df_columns); col_idx++) {
+		LogicalType duckdb_col_type;
+		PandasColumnBindData bind_data;
+
+		names.emplace_back(py::str(df_columns[col_idx]));
+		bind_data.pandas_type = ConvertPandasType(df_types[col_idx]);
+
+		auto column = get_fun(df_columns[col_idx]);
+
+		if (bind_data.pandas_type == PandasType::FLOAT_16) {
+			bind_data.numpy_col = py::array(column.attr("astype")("float32"));
+			bind_data.pandas_type = PandasType::FLOAT_32;
+			duckdb_col_type = PandasToLogicalType(bind_data.pandas_type);
+		} else if (bind_data.pandas_type == PandasType::OBJECT && string(py::str(df_types[col_idx])) == "string") {
+			bind_data.numpy_col =
+			    py::array(column.attr("astype")("object_")); // treat the string array as numpy.object_ array
```

Comment:
What is the reason we don't turn this into an ENUM like we do for strings in BindPandas?

Reply:
Hi Tishj,
  Thanks for the comment. I'm not sure if I get this correctly. But if there is no casting from string ndarrays to object arrays, a segfalut will be triggered at [PyUnicode_CheckExact python api call](https://github.com/vlowingkloude/duckdb-numpy/blob/master/tools/pythonpkg/src/vector_conversion.cpp#L313).

Reply:
Hmm usually that's caused by handling strings without holding the GIL

Or we keep the pointer to these strings around without holding an owning reference, and as a result the strings gst garbage collected.